substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfSectionImpl.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2020, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.elf.dwarf;
+
+import com.oracle.objectfile.BasicProgbitsSectionImpl;
+import com.oracle.objectfile.BuildDependency;
+import com.oracle.objectfile.LayoutDecision;
+import com.oracle.objectfile.LayoutDecisionMap;
+import com.oracle.objectfile.ObjectFile;
+import com.oracle.objectfile.debugentry.ClassEntry;
+import com.oracle.objectfile.elf.ELFObjectFile;
+import org.graalvm.compiler.debug.DebugContext;
+
+import java.nio.ByteOrder;
+import java.util.Map;
+import java.util.Set;
+
+import static com.oracle.objectfile.elf.dwarf.DwarfSections.TEXT_SECTION_NAME;
+
+/**
+ * class from which all DWARF debug sections inherit providing common behaviours.
+ */
+public abstract class DwarfSectionImpl extends BasicProgbitsSectionImpl {
+    protected DwarfSections dwarfSections;
+    public boolean debug = false;
+    public long debugTextBase = 0;
+    public long debugAddress = 0;
+    public int debugBase = 0;
```

Comment:
Is this a write-only field?

Reply:
These are used when trace logging is enabled. They are updated in method enableLog(). n.b. currently only 3 of the DwarfSectionImpl subclasses generate trace. However, I have added the trace log functionality at the superclass level so that all Impl classes use it if/when it turns out to be needed.